### PR TITLE
Adds more unit tests for WCPairing

### PR DESCRIPTION
**Pull request by MaisaMilena**
- Rename some tests to specify if is testing init from uri or topic 
- Add scenarios that throw error 

While working on this I noticed that the `activate` function sets the status to `true` even if `updateExpiry` throws an [error](https://github.com/WalletConnect/WalletConnectSwiftV2/blob/b676369f6dd72d8d0d80fe1275313f1f74cade7b/Sources/WalletConnectPairing/Types/WCPairing.swift#L52). Is that the expected behavior? Maybe `activate` should only change the `active` var?

### CODE REVIEW

**Comment by MaisaMilena** [comment]
Btw, while on `development` running the tests, `CacaoSignerTest` and `SignerTest` failed

**Review by MisterVants — CHANGES_REQUESTED** [review]
Test needs a fix (update less than expiry)

File: Tests/WalletConnectSignTests/WCPairingTests.swift [hunk 1]
```diff
@@ -35,18 +35,61 @@ final class WCPairingTests: XCTestCase {
...
+            XCTAssertEqual(error as! WCPairing.Errors, WCPairing.Errors.invalidUpdateExpiryValue)
+        }
+    }
+
+    func testActivate_WhenUpdateExpiryFails_ShouldActivateAndThrowError() {
+        let expiryDate = referenceDate.advanced(by: 10 * .minute)
+        var pairing = WCPairing(topic: "", relay: .stub(), peerMetadata: .stub(), expiryDate: expiryDate)
+        XCTAssertFalse(pairing.active)
+        pairing.activate()
+        XCTAssertThrowsError(try pairing.updateExpiry(10 * .minute)) { error in
```
Comment (MisterVants) [inline Tests/WalletConnectSignTests/WCPairingTests.swift:90]: `activate()` makes a call to `updateExpiry()` internally (which sets a 30-day expiry). 

Is there a reason to call activate before a throwing update? Does it make a difference for this test to invert the order?

File: Tests/WalletConnectSignTests/WCPairingTests.swift [hunk 2]
```diff
@@ -35,18 +35,61 @@ final class WCPairingTests: XCTestCase {
...
+        var pairing = WCPairing(topic: "", relay: .stub(), peerMetadata: .stub(), expiryDate: referenceDate)
+        XCTAssertThrowsError(try pairing.updateExpiry(40 * .day)) { error in
+            XCTAssertEqual(error as! WCPairing.Errors, WCPairing.Errors.invalidUpdateExpiryValue)
+        }
+    }
+    
+    func testUpdateExpiry_WhenNewExpiryDateIsLessThanExpiryDate_ShouldThrowInvalidUpdateExpiryValue() {
+        let expiryDate = referenceDate.advanced(by: 10 * .minute)
+        var pairing = WCPairing(topic: "", relay: .stub(), peerMetadata: .stub(), expiryDate: expiryDate)
+        XCTAssertThrowsError(try pairing.updateExpiry(40 * .day)) { error in
```
[the remark this answers is not in the document]
Comment (MaisaMilena) [reply Tests/WalletConnectSignTests/WCPairingTests.swift]: ops! 😅

File: Tests/WalletConnectSignTests/WCPairingTests.swift (hunk 1 above)
Comment (MaisaMilena) [reply Tests/WalletConnectSignTests/WCPairingTests.swift]: I was trying to check the error inside the `activate()` function, but if I understood it right the error dies inside `activate()`. I changed the test, now checking for the `expiryDate` update. If an error occurs, it shouldn't be updated.

**Comment by flypaper0** [comment]
lgtm, more tests is always better 👍
fix testcase naming pls "testUpdateExpiry_WhenValueIsGreaterThanMax_ShouldThrowInvalidUpdateExpiryValue"
Little less text and without "_"

**Review by flypaper0 — APPROVED** [review]

**Comment by MaisaMilena** [comment]
@flypaper0 updated! Do these new names fit better?

**Review by MisterVants — APPROVED** [review]

**Pull request merged**